backend/src/modules/cart/service/cart.service.ts

```diff
@@ -31,12 +31,15 @@ export class CartService {
       return newCart.save();
     }
 
-    const update = {
-      userId: cart.userId,
-      items,
-    };
+    await this.cartModel.updateOne(
+      { _id: cart.id },
+      {
+        userId: cart.userId,
+        items,
+      }
+    );
 
-    return this.cartModel.findByIdAndUpdate(cart.id, update).exec();
+    return this.getUserCartItems(userId);
```

Comment:
https://docs.mongodb.com/manual/reference/method/db.collection.findOneAndUpdate

Reply:
returnNewDocument

Reply:
I use `this.getUserCartItems()` because i need to return cart items with price

Reply:
I use `this.getUserCartItems()` because i need to return cart items with price

Reply:
And you return cart items instead of whole cart

Reply:
It can cause some error due to another response format